python/ml4ir/data/tfrecord_reader.py

```diff
@@ -79,72 +88,102 @@ def _parse_sequence_example_fn(sequence_example_proto):
 
             features_dict[feature_node_name] = feature_tensor
 
+        # Define mask to identify padded records
+        if required_only and not feature_config.get_rank("serving_info")["required"]:
+            """
+            Define dummy mask if the rank field is not a required field for serving
+
+            NOTE:
+            This masks all max_num_records as 1 as there is no real way to know
+            the number of records in the query. There is no predefined required field,
+            and hence we would need to do a full pass of all features to find the record shape.
+            This approach might be unstable if different features have different shapes.
```

Comment:
some features *do* have different shapes, right?  Text sequence features, for example.  Have we unit tested if it breaks with that?

Reply:
This is just the mask though. It has a defined shape. `[batch_size, num_records]`. In this particular case, which we will not hit anymore(since we have `rank` defined as a mandatory field at inference time), we just create a mask of 25 1s, because there is no way to know how many records are there.